### ISSUE TITLE
Whitespace problem with latest slim engine

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-reveal.js#8]
The slim template engine changed how it handles whitespace in 3.0.0. The templates need to be upgraded to work with the lastest version of slim.

https://github.com/slim-template/slim/blob/master/CHANGES#L15

**Comment by mojavelinux** [comment]
Here's the warning that's emitted.

```
Deprecated syntax: =' for trailing whitespace is deprecated in favor of =>
  /path/to/reveal.js/templates/slim/inline_menu.html.slim, Line 7, Column 
    ='submenus.map {|submenu| %(<span class="submenu">#{submenu}</span>&#160;&#9656; ) }.join.chop
      ^
```

There may be others.

**Comment by mojavelinux** [comment]
For the time being I documented that Slim 2.1.0 is required to use the templates. See https://github.com/asciidoctor/asciidoctor-reveal.js#usage-instructions

**Comment by obilodeau** [comment]
Will the new syntax work with both the old and the new version? Sorry I
didn't check I'm currently offline.

If it doesn't we will probably want to sync this change with asciidoc
upstream (if it uses slim) so we don't end up with some back ends broken
and others not.

**Comment by jirutka** [comment]
> Will the new syntax work with both the old and the new version?

Of course. Look at [Slim/HTML5](https://github.com/asciidoctor/asciidoctor-backends/tree/master/slim/html5), it doesn’t use the deprecated syntax at all and it works both in 2.1.0 and 3.x.

**Comment by mojavelinux** [comment]
We probably just want to avoid the use of the syntax in question (which I think is true of the slim/html5 templates). That way, we don't have to worry about breaking changes or warnings. There are ways to do without it.

**Comment by mojavelinux** [comment]
The example I cited was the only case, so I fixed it. I'm now using Slim 3, so if I see other issues, I'll address them in turn. Of course, doctest is going to help with this in the long run.

**Comment by jaredmorgs** [comment]
So should this fix be reflected in the README for reveal.js? I think it still has the note about versions present when I took a look tonight.

**Comment by mojavelinux** [comment]
The templates are now compatible with Slim 3.0, but we still recommend using Slim 2.1 until Asciidoctor 1.5.3 is out because you get the deprecation warning about the format option. If the point in the README about the Slim version is not clear enough, we can make it more clear. wdyt?

**Comment by archiloque** [comment]
Hello, Asciidoctor 1.5.3 is out so could we close the topic ?

**Comment by mojavelinux** [comment]
We first need to remove the note about using Slim 2.1.0 in the README. Let's do that as a pull request that resolves this issue, so we get the fix and the close in one shot.

**Comment by mojavelinux** [comment]
Resolved by c6d1b15555b3ae133deebfe0fe656111754ca711. Thanks @archiloque!
